framework/src/modules/chain/dpos/dpos.js

```diff
@@ -77,12 +80,135 @@ module.exports = class Dpos {
 
 	async onRoundFinish() {
 		const disposableDelegateList =
-			this.finalizedBlockRound - this.delegateListRoundOffset;
+			this.finalizedBlockRound -
+			this.delegateListRoundOffset -
+			this.delegateActiveRoundLimit;
 		await this.delegatesList.deleteDelegateListUntilRound(
 			disposableDelegateList,
 		);
 	}
 
+	async getActiveDelegateHeights(
+		numberOfRounds = 1,
+		{ tx, delegateListRoundOffset = this.delegateListRoundOffset } = {},
+	) {
+		const limit =
+			numberOfRounds + this.delegateActiveRoundLimit + delegateListRoundOffset;
```

Comment:
shouldn't this be `-1`?
`1 + 3 + 2` in the numberOfRounds = 1 case, but it should be 5?